runtime/server/catalog.go

```diff
@@ -33,6 +33,58 @@ func (s *Server) ListCatalogEntries(ctx context.Context, req *runtimev1.ListCata
 		return nil, status.Error(codes.Unknown, err.Error())
 	}
 
+	if req.Type == runtimev1.ObjectType_OBJECT_TYPE_METRICS_VIEW {
+		// apply access policies to metrics views
+		// check metrics view access and column access
+		accessibleEntries := make([]*drivers.CatalogEntry, 0)
+
+		for _, entry := range entries {
+			mv := entry.GetMetricsView()
+			mv.CatalogUpdatedOn = timestamppb.New(entry.UpdatedOn)
+
+			policy, err := auth.ResolveMetricsViewPolicy(auth.GetClaims(ctx).UserClaims(), mv)
+			if err != nil {
+				return nil, err
+			}
+
+			if policy != nil {
+				if !policy.HasAccess {
+					continue
+				}
+				accessibleEntries = append(accessibleEntries, entry)
+
+				allowedDims := make([]*runtimev1.MetricsView_Dimension, 0)
+
+				if len(policy.Include) > 0 {
+					allowed := make(map[string]bool)
+					for _, include := range policy.Include {
+						allowed[include] = true
+					}
+					for _, dim := range mv.Dimensions {
+						if allowed[dim.Name] {
+							allowedDims = append(allowedDims, dim)
+						}
+					}
+					mv.Dimensions = allowedDims
+				} else if len(policy.Exclude) > 0 {
+					restricted := make(map[string]bool)
+					for _, exclude := range policy.Exclude {
+						restricted[exclude] = true
+					}
+					for _, dim := range mv.Dimensions {
+						if !restricted[dim.Name] {
+							allowedDims = append(allowedDims, dim)
+						}
+					}
+					mv.Dimensions = allowedDims
+				}
```

Comment:
1. Maybe pull it into a separate function the same logic is in `GetCatalogEntry`
2. I'm not sure it's safe to mutate the `mv` returned from a catalog lookup, might be cached now or in the future

Reply:
2. This is just a GET API call and seriliazed entry is returned to the caller so I think it should be safe, caching may happen at DB layer but we are mutating before returning the entry. Any other case that you can think of ?

Reply:
We are adding a catalog object cache in async reconcile, so it won't be safe to mutate the returned values without copying it